### ISSUE TITLE
Inline fields editing fails with "RuntimeError: OrderedDict mutated during iteration" on CPython 3.5

### ISSUE THREAD

**Issue by lopuhin** [flask-admin/flask-admin#1283]
Inline fields editing fails with `RuntimeError: OrderedDict mutated during iteration` on CPython 3.5, using `Flask-Admin==1.4.0`. Here is the traceback:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.5/site-packages/flask_admin/base.py", line 68, in inner
    return self._run_view(f, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/flask_admin/base.py", line 367, in _run_view
    return fn(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/flask_admin/model/base.py", line 2082, in ajax_update
    for field in form:
RuntimeError: OrderedDict mutated during iteration
```

**Comment by flobr** [comment]
the problem is that in 

    def ajax_update(self):
        """
            Edits a single column of a record in list view.
        """
        if not self.column_editable_list:
            abort(404)

        record = None
        form = self.list_form()

        # prevent validation issues due to submitting a single field
        # delete all fields except the field being submitted
        for field in form:
            # only the submitted field has a positive last_index
            if getattr(field, 'last_index', 0):
                record = self.get_one(str(field.last_index))
            elif field.name == 'csrf_token':
                pass
            else:
                form.__delitem__(field.name)

the form which seems to use an OrderedDict is modified directly while it is iterated. An easy way to get it working is to override the method in you own view and instead of deleting the fields directly first store their names in an array. Then loop over your temp array and remove the fields from the form.

**Comment by eloyz** [comment]
@flobr you're solution helps for now.  Thank you guys.

**Comment by tandreas** [comment]
This was fixed by commit d940c21da7466aed5442174a88cbcd434382e7f2 and should be closed.
